### PR TITLE
Rename js feature to web

### LINKED ISSUE #3844: Rename `js` feature to `web`
Rename the `wasmer` crate `js` feature to `web` (we should still support `js` feature, but it should just depend on `web`.

### PR DESCRIPTION
**Pull request by syrusakbary**
JS feature might be a bit confusing, so we decided to rename the `js` feature to `web` (we'll keep supporting `js` for now for compatibility reasons).

Fixes #3844

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -109,14 +109,14 @@ jobs:
...
           curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh
-      - name: make test-js
+      - name: make test-web
         run: |
-          make test-js
+          make test-web
 
       # The no_std functionality doesn't work at the moment - no point in testing it.
-      # - name: make test-js-core
+      # - name: make test-web-core
```
Comment (ptitSeb) [inline .github/workflows/test.yaml:117]: web-core? we don't keep js-core here?

File: lib/api/src/lib.rs [hunk 2]
```diff
@@ -451,10 +451,10 @@ mod sys;
 #[cfg(feature = "sys")]
 pub use sys::*;
 
-#[cfg(feature = "js")]
+#[cfg(feature = "web")]
 mod js;
```
Comment (ptitSeb) [inline lib/api/src/lib.rs:455]: the feature is web but the mod stay `js`?

**Review by ptitSeb — APPROVED** [review]

File: lib/api/src/lib.rs (hunk 2 above)
Comment (syrusakbary) [reply lib/api/src/lib.rs:455]: Good call. Let me fix that

**Review by syrusakbary — COMMENTED** [review]
DO NOT MERGE YET. I'm wondering if web is a good name when Wasmer get's compiled for Node.js

**Pull request closed without merging**